WH linker error when building BRISC with watcher enabled

I'm trying to get the stress test pipeline error-free again and I'm seeing a regression issue pop up with watcher enabled. To reproduce: 

```
TT_METAL_WATCHER=30 ./build/test/tt_metal/unit_tests_fast_dispatch
```

Error signature:

```
In file included from ./kernel.cpp:5,
                 from /home/ubuntu/bteng/tt-metal/tt_metal/hw/firmware/src/brisck.cc:23:
/home/ubuntu/bteng/tt-metal/tt_metal/impl/dispatch/kernels/cq_prefetcher.hpp: In function 'void completion_queue_reserve_back(uint32_t)':
/home/ubuntu/bteng/tt-metal/tt_metal/impl/dispatch/kernels/cq_prefetcher.hpp:153:80: warning: suggest parentheses around '&&' within '||' [-Wparentheses]
  153 |         (completion_rd_toggle != cq_write_interface.completion_fifo_wr_toggle) and (cq_write_interface.completion_fifo_wr_ptr == completion_rd_ptr)
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/ubuntu/bteng/tt-metal/tt_metal/third_party/sfpi/compiler/bin/../lib/gcc/riscv32-unknown-elf/10.2.0/../../../../riscv32-unknown-elf/bin/ld: /home/ubuntu/bteng/tt-metal/built/1/kernels/cq_prefetcher/17186141426769429586/brisc/brisc.elf section `.text' will not fit in region `BRISC_CODE'
/home/ubuntu/bteng/tt-metal/tt_metal/third_party/sfpi/compiler/bin/../lib/gcc/riscv32-unknown-elf/10.2.0/../../../../riscv32-unknown-elf/bin/ld: region `BRISC_CODE' overflowed by 1048 bytes
collect2: error: ld returned 1 exit status
                 Always | FATAL    | brisc build failed
                 Always | FATAL    | Failed to generate binaries for cq_prefetcher TT_THROW @ tt_metal/jit_build/build.cpp:372: tt::exception
info:
{} build failed
brisc
backtrace:
 --- tt::tt_metal::JitBuildState::link(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const
 --- tt::tt_metal::JitBuildState::build(tt::tt_metal::JitBuildSettings const*) const
 --- /home/ubuntu/bteng/tt-metal/build/lib/libtt_metal.so(+0x2976ea) [0x7f93e78fc6ea]
 --- /home/ubuntu/bteng/tt-metal/build/lib/libtt_metal.so(+0x2979f4) [0x7f93e78fc9f4]
 --- /home/ubuntu/bteng/tt-metal/build/lib/libtt_metal.so(+0x29fc2d) [0x7f93e7904c2d]
 --- /lib/x86_64-linux-gnu/libpthread.so.0(+0x114df) [0x7f93e76464df]
 --- /home/ubuntu/bteng/tt-metal/build/lib/libtt_metal.so(+0x28fa3c) [0x7f93e78f4a3c]
 --- /home/ubuntu/bteng/tt-metal/build/lib/libtt_metal.so(+0x2af187) [0x7f93e7914187]
 --- std::thread::_State_impl<std::thread::_Invoker<std::tuple<tf::Executor::_spawn(unsigned long)::{lambda()#1}> > >::_M_run()
 --- /lib/x86_64-linux-gnu/libstdc++.so.6(+0xd6df4) [0x7f93e7529df4]
 --- /lib/x86_64-linux-gnu/libpthread.so.0(+0x8609) [0x7f93e763d609]
 --- /lib/x86_64-linux-gnu/libc.so.6(clone+0x43) [0x7f93e7088353]
```

--- a/tt_metal/hw/inc/dataflow_api.h
+++ b/tt_metal/hw/inc/dataflow_api.h
@@ -29,8 +29,6 @@
 
 extern uint8_t noc_index;
 
-#define FORCE_INLINE inline __attribute__((always_inline))
-
 /** @file */
 
 /**

--- a/tt_metal/hw/inc/ethernet/tunneling.h
+++ b/tt_metal/hw/inc/ethernet/tunneling.h
@@ -12,8 +12,6 @@
 #include "tt_metal/impl/dispatch/kernels/command_queue_common.hpp"
 #include "tt_metal/impl/dispatch/kernels/cq_prefetcher.hpp"
 
-#define FORCE_INLINE inline __attribute__((always_inline))
-
 inline void RISC_POST_STATUS(uint32_t status) {
     volatile uint32_t *ptr = (volatile uint32_t *)(NOC_CFG(ROUTER_CFG_2));
     ptr[0] = status;

--- a/tt_metal/hw/inc/risc_attribs.h
+++ b/tt_metal/hw/inc/risc_attribs.h
@@ -37,4 +37,10 @@ inline __attribute__((always_inline)) uint64_t tt_l1_load(volatile tt_uint64_t *
     return v.v;
 }
 
+#if not defined(WATCHER_ENABLED)
+    #define FORCE_INLINE inline __attribute__((always_inline))
+#else
+    #define FORCE_INLINE
+#endif
+
 #endif // _RISC_ATTRIBS_H_

--- a/tt_metal/include/compute_kernel_api/common.h
+++ b/tt_metal/include/compute_kernel_api/common.h
@@ -10,8 +10,6 @@
 #include "compute_kernel_api/unpack.h"
 #include "compute_kernel_api/cb_api.h"
 
-#define FORCE_INLINE inline __attribute__((always_inline))
-
 /**
  * Returns the address in L1 for a given runtime argument index
  *

